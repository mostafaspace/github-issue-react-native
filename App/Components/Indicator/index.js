import React from 'react';
import {ActivityIndicator} from 'react-native';
import PropTypes from 'prop-types';
import Styles from './styles';
import {Colors} from '../../Constants';

export const Indicator = ({size, containerStyle}) => {
    return (
        <ActivityIndicator
            size={size ? size : 'large'}
            color={Colors.grey}
            style={[Styles.indicator, containerStyle]}
            testID={'Indicator'}
        />
    );
};

Indicator.propTypes = {
    size: PropTypes.string,
    containerStyle: PropTypes.object,
};
