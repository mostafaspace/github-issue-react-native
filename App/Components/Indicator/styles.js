import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    indicator: {
        flex: 1,
        justifyContent: 'center',
        alignSelf: 'center',
    },
});
