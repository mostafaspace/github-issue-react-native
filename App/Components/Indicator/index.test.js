import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import {Indicator} from '../Indicator';
import renderer from 'react-test-renderer';
import '@testing-library/jest-native/extend-expect';

afterEach(cleanup);
it('renders component', () => {
    render(<Indicator />);
});

it('Indicator matches snapshot', () => {
    const tree = renderer.create(<Indicator text={'test'} />).toJSON()
    expect(tree).toMatchSnapshot();
});
