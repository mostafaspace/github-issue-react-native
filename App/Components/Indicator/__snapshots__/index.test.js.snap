// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator matches snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#575757"
  hidesWhenStopped={true}
  size="large"
  style={
    Array [
      Object {
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="Indicator"
/>
`;
